public/scripts/ssooverride.js

```diff
@@ -20,110 +20,76 @@
-
-      $('.dropdown').each(function (index, el) {
-        var dropDownMenu = el.querySelector('.dropdown-menu');
-        var dropDownToggle = el.querySelector('.dropdown-toggle');
-        dropDownToggle.addEventListener('click', function (e) {
-          e.preventDefault();
-          if (dropDownMenu.style.display === 'block') {
-            dropDownMenu.style.display = '';
-          } else {
-            dropDownMenu.style.display = 'block';
-          }
-        }, false);
-      });
-
-      var basicQuery = "?" + [
-        "client_id=" + oauthClientId,
-        "response_type=code",
-        "scopes=user email"
+    var joinEl = $('#signup-link');
+    var loginEl = $('#login-link');
+
+    function enable(user) {
+      // Logic for UX on detection of a logged in user should
+      // go here
+    };
+
+    function disable() {
+      // The default state.
+    }
+
+    var basicQuery = "?" + [
+      "client_id=" + oauthClientId,
+      "response_type=code",
+      "scopes=user email"
+    ].join("&");
+
+    // Signup login flow
+    joinEl.on('click', function(e) {
+      e.preventDefault();
+
+      var state = uuid.v4();
+      var oauthRoute = "/login/oauth/authorize";
+
+      var query = authHostname + oauthRoute + basicQuery + [
+        "&state=" + state,
+        "action=signup"
+      ].join("&");
+
+      setStateCookie(state);
+
+      window.location = query;
+    });
+    // Login flow
+    loginEl.on('click', function(e) {
+      e.preventDefault();
+
+      var state = uuid.v4();
+      var oauthRoute = "/login/oauth/authorize";
+
+      var query = authHostname + oauthRoute + basicQuery + [
+        "&state=" + state,
+        "action=signin"
       ].join("&");
 
-      // Signup login flow
-      joinEl.on('click', function() {
-        var state = uuid.v4();
-        var oauthRoute = "/login/oauth/authorize";
-
-        var query = authHostname + oauthRoute + basicQuery + [
-          "&state=" + state,
-          "action=signup"
-        ].join("&");
-
-        setStateCookie(state);
-
-        window.location = query;
-      });
-      // Login flow
-      loginEl.on('click', function() {
-        var state = uuid.v4();
-        var oauthRoute = "/login/oauth/authorize";
-
-        var query = authHostname + oauthRoute + basicQuery + [
-          "&state=" + state,
-          "action=signin"
-        ].join("&");
-
-        setStateCookie(state);
-
-        window.location = query;
-      });
-      logoutEl.on('click', function() {
-        // Logout flow
-        var oauthRoute = "/logout";
-
-        var query = authHostname + oauthRoute + "?client_id=" + oauthClientId;
-
-        window.location = query;
-      });
-
-      if (user.username.length > 0) {
-        enable(user);
-      } else {
-        // We disable the publishing UI by default
-        disable();
-      }
+      setStateCookie(state);
+
+      window.location = query;
     });
+
+    // XXXBramble - This callback must execute on logout:
```

Comment:
Yes, since we don't have UI for logout yet. I'll update the comment to be a little clearer.